### ISSUE TITLE
DeleteMarkLogic output varies

### ISSUE THREAD

**Issue by eouthwaite** [marklogic/nifi#74]
I have 669 documents in a pair of collections whose name is stored in the variable ${staging_collections} (`evaluation, nifi`) that I wish to delete

I have 2 issues with the DeleteMarkLogic processor:

1. I would prefer a single "success" output flowfile rather than a flowfile for each deleted document
2. The output varies based on how I set the deletion:
* using Collection Query:
  * using batch size 100, depending on the run, I get 585-600 flow files and all 669 documents are deleted
  * batch size 1000, I get 500 flow files and all 669 documents are deleted
* using Structured Query XML and the following query:
```<cts:and-query xmlns:cts="http://marklogic.com/cts">
    <cts:collection-query xmlns:cts="http://marklogic.com/cts">
        <cts:uri>${staging_collections:substringBefore(',')}</cts:uri>
    </cts:collection-query>
    <cts:collection-query xmlns:cts="http://marklogic.com/cts">
        <cts:uri>${staging_collections:substringAfter(', ')}</cts:uri>
    </cts:collection-query>
</cts:and-query>
```

  * using batch size 1000, 1 or 3 threads, I get 500 flow files and all 669 documents are deleted
  * batch size 100, I get 502 flow files and all 669 documents are deleted
  * batch size 1, I get 661 flow files and all 669 documents are deleted

**Comment by rjrudin** [comment]
I think we need to provide an option of what DeleteML returns, with the single FF (including a count of deleted URIs) being the default. It seems rare that a client will need to know the URI of every deleted document, much less get a FF for every one, but we can still provide that as an option.

**Comment by rjrudin** [comment]
I think the design change of returning one FF per batch or for all deleted documents is out of scope here, particularly as that would be a backwards incompatibility. 

I believe the issue with not getting the right number of flowfiles back has been fixed by #91, where there was definitely a problem with the first batch of flowfiles failed to be committed due to the original flowfile not having a destination set on it.
